### PR TITLE
[fix] Fixed traffic charts showing wrong traffic total #415

### LINKED ISSUE #415: [bug] Traffic total sometimes is very wrong
This is happening in the current development version (1.1.0 alpha).

Screenshot from 2022-07-15 17-05-23

The total amount in some cases does not match the download + upload exactly, but it should.

### PR DESCRIPTION
**Pull request by pandafy**
The traffic charts were showing the wrong total because of the
approximation (round-off) done in the backend to decrease the
the precision of values to two decimal places.

Since the adaptive feature already does this approximation
in the frontend, the rounding off is skipped from the backend.

Closes #415

Checks:

- [x] I have manually tested the proposed changes
- [x] I have written new test cases to avoid regressions (if necessary)
- [ ] I have updated the documentation (e.g. README.rst)

### CODE REVIEW

File: openwisp_monitoring/monitoring/base/models.py [hunk 1]
```diff
@@ -586,7 +586,14 @@ def read(
                 if key == 'time':
                     continue
                 traces.setdefault(key, [])
-                if decimal_places and isinstance(value, (int, float)):
+                if (
+                    # The adaptive feature does unit conversion and approximation
+                    # in the frontend, hence rounding off is skipped here.
+                    # TODO: Update this code to check for 'adaptive_prefix'
+                    not self.config_dict.get('unit', '').startswith('adaptive')
```
Comment (nemesisdesign) [inline openwisp_monitoring/monitoring/base/models.py:593]: why this TODO here?

**Review by nemesisdesign — COMMENTED** [review]
Looks good, see my question below.

File: openwisp_monitoring/monitoring/base/models.py (hunk 1 above)
Comment (pandafy) [reply openwisp_monitoring/monitoring/base/models.py:593]: In `gsoc22-demo` PR, Aryaman has changed `adaptive_bytes` to `adaptive_prefix`. The current code with work with both `adaptive_bytes` and `adaptive_prefix`. But, later on we might choose to use `adaptive_prefix` only.

Comment (nemesisdesign) [reply openwisp_monitoring/monitoring/base/models.py:593]: ok clear now, thanks! We shall create an issue for this to remind ourselves.

**Review by nemesisdesign — APPROVED** [review]
Looks good! Let's keep testing it a bit more.

**Pull request merged**